packages/app/src/vis-packs/nexus/NxValuesFetcher.tsx

```diff
@@ -30,13 +33,26 @@ function NxValuesFetcher<T extends NumericType | ComplexType>(props: Props<T>) {
   usePrefetchValues([signalDataset, errorsDataset, ...auxDatasets], dimMapping);
   usePrefetchValues([...axisDatasets, titleDataset]);
 
+  const { attrValuesStore } = useContext(ProviderContext);
   const signal = useDatasetValue(signalDataset, dimMapping);
+  const signalLabel = getDatasetLabel(signalDataset, attrValuesStore);
   const errors = useDatasetValue(errorsDataset);
   const axisMapping = useAxisMapping(axisDatasets, silxStyle.axisScaleTypes);
   const auxiliaries = Object.values(useDatasetValues(auxDatasets));
-  const title = useDatasetValue(titleDataset);
+  const title = useDatasetValue(titleDataset) || signalLabel;
```

Comment:
The visualization title is now determined and provided by the `NxValuesFetcher`. The visualizations no longer have to write `title || getDatasetLabel(...)` (which would be more annoying now that we have to pass `attrValuesStore` to `getDatasetLabel`...)

We have to always compute the `signalLabel` and provide it, as some visualizations make use of it regardless of the present of `title`. The (acceptable) downside of this refactoring is that if an NXdata group has a `title` dataset AND a signal dataset with a `unit` or `long_name`, then the value of one of these attributes will be fetched even though it will not be used. Nothing critical...